pkg/gui/controllers/files_controller.go

```diff
@@ -748,6 +755,97 @@ func (self *FilesController) createStashMenu() error {
 	})
 }
 
+func (self *FilesController) openCopyMenu() error {
+	copyNameNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "name"},
+	)
+	copyDiffNoItemSelectionErr := utils.ResolvePlaceholderString(
+		self.c.Tr.FileCopyNoItemSelectedError,
+		map[string]string{"target": "diff"},
+	)
+	copyNameItem := &types.MenuItem{
+		Label: self.c.Tr.CopyFileName,
+		OnPress: func() error {
+			name := self.context().GetSelectedItemId()
+			if name == "" {
+				return self.c.ErrorMsg(copyNameNoItemSelectionErr)
+			}
+
+			return self.copyToClipboard(name)
+		},
+		Key: 'n',
+	}
+	copyFileDiffItem := &types.MenuItem{
+		Label: self.c.Tr.CopySelectedDiff,
+		OnPress: func() error {
+			path := self.context().GetSelectedPath()
+			if path == "" {
+				return self.c.ErrorMsg(copyDiffNoItemSelectionErr)
+			}
+			diff, err := self.c.Git().Diff.GetPathDiff(path, self.hasPathStagedChanges())
+			if err != nil {
+				return self.c.Error(err)
+			}
+			return self.copyToClipboard(diff)
+		},
+		Key: 's',
+	}
+
+	if self.context().GetSelected() == nil {
+		copyNameItem.DisabledReason = copyNameNoItemSelectionErr
+		copyFileDiffItem.DisabledReason = copyDiffNoItemSelectionErr
+	}
+
+	return self.c.Menu(types.CreateMenuOptions{
+		Title: self.c.Tr.CopyToClipboardMenu,
+		Items: []*types.MenuItem{
+			copyNameItem,
+			copyFileDiffItem,
+			{
+				Label: self.c.Tr.CopyAllFilesDiff,
+				OnPress: func() error {
+					diff, err := self.c.Git().Diff.GetAllDiff(self.hasStagedChanges())
```

Comment:
Why not just use `self.getSelectedFile().HasStagedChanges` ?

Reply:
I'll use the `self.c.Helpers().WorkingTree.AnyStagedFiles()` method here, as you suggested. Did you mean to use the `self.getSelectedFile().HasStagedChanges` on line [786](https://github.com/jesseduffield/lazygit/pull/3104/files#diff-0776a04e9be3fc3fc0a681b9b122a1375858afc8de884062866b8a68fbf51692R786) instead?